### PR TITLE
Functional tests - Add new test 'BO > Product V2 > Pricing tab'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add new test 'BO > Product V2 > Pricing tab'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | no need
| UI Tests          | Nightly tests => https://github.com/nesrineabdmouleh/testing_pr/actions/runs/6313569378
| Fixed issue or discussion?     | no
| Related PRs       | no
| Sponsor company   |

### CODE REVIEW

File: tests/UI/pages/BO/catalog/productsV2/add/pricingTab.ts [hunk 1]
```diff
@@ -4,6 +4,8 @@ import BOBasePage from '@pages/BO/BObasePage';
 // Import data
 import ProductData from '@data/faker/product';
 
+import type {ProductSpecificPrice} from '@data/types/product';
+
```
Comment (Progi1984) [inline tests/UI/pages/BO/catalog/productsV2/add/pricingTab.ts]:
```suggestion
import type {ProductSpecificPrice} from '@data/types/product';

```

File: tests/UI/campaigns/productV2/functional/17_pricingTab.ts [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+
+    it('should go to \'Catalog > Products\' page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToProductsPage', baseContext);
+
+      await dashboardPage.goToSubMenu(
+        page,
+        dashboardPage.catalogParentLink,
+        dashboardPage.productsLink,
+      );
+
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/17_pricingTab.ts]:
```suggestion
```

File: tests/UI/campaigns/productV2/functional/17_pricingTab.ts [hunk 3]
```diff
@@ -0,0 +1,532 @@
...
+        expect(result.marginRateValue).to.eq('100.00% margin rate'),
+        expect(result.WholesalePriceValue).to.eq('€0.00 cost price'),
+      ]);
+    });
+
+    it('should preview product', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'previewProduct', baseContext);
+
+      page = await createProductPage.previewProduct(page);
+
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/17_pricingTab.ts]:
```suggestion
```

File: tests/UI/campaigns/productV2/functional/17_pricingTab.ts [hunk 4]
```diff
@@ -0,0 +1,532 @@
...
+        expect(result.marginRateValue).to.eq('100.00% margin rate'),
+        expect(result.WholesalePriceValue).to.eq('€0.00 cost price'),
+      ]);
+    });
+
+    it('should preview product', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'previewProduct2', baseContext);
+
+      page = await createProductPage.previewProduct(page);
+
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/17_pricingTab.ts]:
```suggestion
```

File: tests/UI/campaigns/productV2/functional/17_pricingTab.ts [hunk 5]
```diff
@@ -0,0 +1,532 @@
...
+        expect(result.marginRateValue).to.eq('65.00% margin rate'),
+        expect(result.WholesalePriceValue).to.eq('€35.00 cost price'),
+      ]);
+    });
+
+    it('should edit Retail price per unit section', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'editRetailPriceParUnit', baseContext);
+
+      await pricingTab.setDisplayRetailPricePerUnit(page, true);
+
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/17_pricingTab.ts]:
```suggestion
```

File: tests/UI/campaigns/productV2/functional/17_pricingTab.ts [hunk 6]
```diff
@@ -0,0 +1,532 @@
...
+      const unitPrice = await pricingTab.getUnitPriceValue(page);
+      expect(unitPrice).to.eq('€10.00 per unit');
+    });
+
+    it('should preview product', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'previewProduct3', baseContext);
+
+      page = await createProductPage.previewProduct(page);
+
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/17_pricingTab.ts]:
```suggestion
```

File: tests/UI/campaigns/productV2/functional/17_pricingTab.ts [hunk 7]
```diff
@@ -0,0 +1,532 @@
...
+      const message = await createProductPage.saveProduct(page);
+      expect(message).to.eq(createProductPage.successfulUpdateMessage);
+    });
+
+    it('should preview product', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'previewProduct4', baseContext);
+
+      page = await createProductPage.previewProduct(page);
+
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/17_pricingTab.ts]:
```suggestion
```

File: tests/UI/campaigns/productV2/functional/17_pricingTab.ts [hunk 8]
```diff
@@ -0,0 +1,532 @@
...
+      const message = await pricingTab.setSpecificPrice(page, specificPriceData.specificPrice);
+      expect(message).to.equal(createProductPage.successfulCreationMessage);
+    });
+
+    it('should preview product', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'previewProduct5', baseContext);
+
+      page = await createProductPage.previewProduct(page);
+
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/17_pricingTab.ts]:
```suggestion
```

File: tests/UI/campaigns/productV2/functional/17_pricingTab.ts [hunk 9]
```diff
@@ -0,0 +1,532 @@
...
+      const message = await pricingTab.setSpecificPrice(page, editSpecificPriceData.specificPrice);
+      expect(message).to.equal('Update successful');
+    });
+
+    it('should preview product', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'previewProduct6', baseContext);
+
+      page = await createProductPage.previewProduct(page);
+
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/17_pricingTab.ts]:
```suggestion
```

File: tests/UI/campaigns/productV2/functional/17_pricingTab.ts [hunk 10]
```diff
@@ -0,0 +1,532 @@
...
+      const successMessage = await pricingTab.deleteSpecificPrice(page, 1);
+      expect(successMessage).to.eq(pricingTab.successfulDeleteMessage);
+    });
+
+    it('should preview product', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'previewProduct7', baseContext);
+
+      page = await createProductPage.previewProduct(page);
+
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/17_pricingTab.ts]:
```suggestion
```

File: tests/UI/campaigns/productV2/functional/17_pricingTab.ts [hunk 11]
```diff
@@ -0,0 +1,532 @@
...
+      const validationMessage = await createCatalogPriceRulePage.setCatalogPriceRule(page, newCatalogPriceRuleData);
+      expect(validationMessage).to.contains(catalogPriceRulesPage.successfulCreationMessage);
+    });
+
+    it('should close the page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'closePage', baseContext);
+
+      page = await catalogPriceRulesPage.closePage(browserContext, page, 0);
+
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/17_pricingTab.ts]:
```suggestion
```

**Review by Progi1984 — COMMENTED** [review]
@nesrineabdmouleh : Thank you for your contribution. Some small feedback.

**Review by Progi1984 — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by boubkerbribri** [comment]
Thanks @nesrineabdmouleh

**Pull request merged**